{"cells":[{"cell_type":"markdown","metadata":{},"source":["![image](https://github.com/IBM/watson-machine-learning-samples/raw/master/cloud/notebooks/headers/AutoAI-Banner_Experiment-Notebook.png)\n","# Experiment Notebook - AutoAI Notebook v1.15.0\n","\n"]},{"cell_type":"markdown","metadata":{},"source":["\n","## Contents\n","\n","This notebook contains the following parts:\n","\n","**[Setup](#setup)**<br>\n","&nbsp;&nbsp;[Package installation](#install)<br>\n","&nbsp;&nbsp;[Watson Machine Learning connection](#connection)<br>\n","**[Experiment configuration](#configuration)**<br>\n","&nbsp;&nbsp;[Experiment metadata](#metadata)<br>\n","**[Working with completed AutoAI experiment](#work)**<br>\n","&nbsp;&nbsp;[Get fitted AutoAI optimizer](#get)<br>\n","&nbsp;&nbsp;[Pipelines comparison](#comparison)<br>\n","&nbsp;&nbsp;[Get pipeline as scikit-learn pipeline model](#get_pipeline)<br>\n","&nbsp;&nbsp;[Inspect pipeline](#inspect_pipeline)<br>\n","&nbsp;&nbsp;&nbsp;&nbsp;[Visualize pipeline model](#visualize)<br>\n","&nbsp;&nbsp;&nbsp;&nbsp;[Preview pipeline model as python code](#preview)<br>\n","**[Deploy and Score](#scoring)**<br>\n","&nbsp;&nbsp;[Working with spaces](#working_spaces)<br>\n","**[Running AutoAI experiment with Python SDK](#run)**<br>\n","**[Clean up](#cleanup)**<br>\n","**[Next steps](#next_steps)**<br>\n","**[Copyrights](#copyrights)**"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"setup\"></a>\n","# Setup"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"install\"></a>\n","## Package installation\n","Before you use the sample code in this notebook, install the following packages:\n"," - ibm_watson_machine_learning,\n"," - autoai-libs,\n"," - lale,\n"," - scikit-learn,\n"," - xgboost,\n"," - lightgbm.\n"]},{"cell_type":"code","execution_count":5,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[{"name":"stdout","output_type":"stream","text":["Collecting ibm-watson-machine-learning\n","  Using cached ibm_watson_machine_learning-1.0.253-py3-none-any.whl (1.8 MB)\n","Collecting ibm-cos-sdk==2.12.*\n","  Using cached ibm_cos_sdk-2.12.0-py3-none-any.whl\n","Collecting importlib-metadata\n","  Using cached importlib_metadata-5.0.0-py3-none-any.whl (21 kB)\n","Collecting certifi\n","  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)\n","Requirement already satisfied: packaging in c:\\users\\dantu\\appdata\\roaming\\python\\python310\\site-packages (from ibm-watson-machine-learning) (21.3)\n","Requirement already satisfied: urllib3 in c:\\python310\\lib\\site-packages (from ibm-watson-machine-learning) (1.26.12)\n","Requirement already satisfied: tabulate in c:\\python310\\lib\\site-packages (from ibm-watson-machine-learning) (0.9.0)\n","Collecting pandas<1.5.0,>=0.24.2\n","  Using cached pandas-1.4.4-cp310-cp310-win_amd64.whl (10.0 MB)\n","Collecting requests\n","  Using cached requests-2.28.1-py3-none-any.whl (62 kB)\n","Requirement already satisfied: lomond in c:\\python310\\lib\\site-packages (from ibm-watson-machine-learning) (0.3.3)\n","Collecting ibm-cos-sdk-core==2.12.0\n","  Using cached ibm_cos_sdk_core-2.12.0-py3-none-any.whl\n","Collecting ibm-cos-sdk-s3transfer==2.12.0\n","  Using cached ibm_cos_sdk_s3transfer-2.12.0-py3-none-any.whl\n","Requirement already satisfied: jmespath<1.0.0,>=0.10.0 in c:\\python310\\lib\\site-packages (from ibm-cos-sdk==2.12.*->ibm-watson-machine-learning) (0.10.0)\n","Requirement already satisfied: python-dateutil<3.0.0,>=2.8.2 in c:\\users\\dantu\\appdata\\roaming\\python\\python310\\site-packages (from ibm-cos-sdk-core==2.12.0->ibm-cos-sdk==2.12.*->ibm-watson-machine-learning) (2.8.2)\n","Requirement already satisfied: pytz>=2020.1 in c:\\python310\\lib\\site-packages (from pandas<1.5.0,>=0.24.2->ibm-watson-machine-learning) (2022.4)\n","Requirement already satisfied: numpy>=1.21.0 in c:\\python310\\lib\\site-packages (from pandas<1.5.0,>=0.24.2->ibm-watson-machine-learning) (1.23.4)\n","Collecting idna<4,>=2.5\n","  Using cached idna-3.4-py3-none-any.whl (61 kB)\n","Collecting charset-normalizer<3,>=2\n","  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)\n","Requirement already satisfied: zipp>=0.5 in c:\\python310\\lib\\site-packages (from importlib-metadata->ibm-watson-machine-learning) (3.9.0)\n","Requirement already satisfied: six>=1.10.0 in c:\\users\\dantu\\appdata\\roaming\\python\\python310\\site-packages (from lomond->ibm-watson-machine-learning) (1.16.0)\n","Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in c:\\users\\dantu\\appdata\\roaming\\python\\python310\\site-packages (from packaging->ibm-watson-machine-learning) (3.0.9)\n","Installing collected packages: importlib-metadata, idna, charset-normalizer, certifi, requests, pandas, ibm-cos-sdk-core, ibm-cos-sdk-s3transfer, ibm-cos-sdk, ibm-watson-machine-learning\n"]},{"name":"stderr","output_type":"stream","text":["WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","  WARNING: Failed to write executable - trying to use .deleteme logic\n","ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 'C:\\\\Python310\\\\Scripts\\\\normalizer.exe' -> 'C:\\\\Python310\\\\Scripts\\\\normalizer.exe.deleteme'\n","\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","ERROR: Could not find a version that satisfies the requirement autoai-libs==1.12.7 (from versions: 1.14.0, 1.14.2, 1.14.3, 1.14.4, 1.14.6)\n","ERROR: No matching distribution found for autoai-libs==1.12.7\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","The system cannot find the file specified.\n"]},{"name":"stdout","output_type":"stream","text":["Collecting scikit-learn==0.23.2\n","  Using cached scikit-learn-0.23.2.tar.gz (7.2 MB)\n","  Installing build dependencies: started\n","  Installing build dependencies: finished with status 'error'\n"]},{"name":"stderr","output_type":"stream","text":["WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","  error: subprocess-exited-with-error\n","  \n","  × pip subprocess to install build dependencies did not run successfully.\n","  │ exit code: 1\n","  ╰─> [620 lines of output]\n","      WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","      Ignoring numpy: markers 'python_version == \"3.6\" and platform_system != \"AIX\" and platform_python_implementation == \"CPython\"' don't match your environment\n","      Ignoring numpy: markers 'python_version == \"3.6\" and platform_system != \"AIX\" and platform_python_implementation != \"CPython\"' don't match your environment\n","      Ignoring numpy: markers 'python_version == \"3.7\" and platform_system != \"AIX\"' don't match your environment\n","      Ignoring numpy: markers 'python_version == \"3.6\" and platform_system == \"AIX\"' don't match your environment\n","      Ignoring numpy: markers 'python_version == \"3.7\" and platform_system == \"AIX\"' don't match your environment\n","      Ignoring numpy: markers 'python_version >= \"3.8\" and platform_system == \"AIX\"' don't match your environment\n","      Collecting setuptools\n","        Using cached setuptools-65.5.0-py3-none-any.whl (1.2 MB)\n","      Collecting wheel\n","        Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)\n","      Collecting Cython>=0.28.5\n","        Using cached Cython-0.29.32-py2.py3-none-any.whl (986 kB)\n","      Collecting numpy==1.17.3\n","        Using cached numpy-1.17.3.zip (6.4 MB)\n","        Preparing metadata (setup.py): started\n","        Preparing metadata (setup.py): finished with status 'done'\n","      Collecting scipy>=0.19.1\n","        Using cached scipy-1.9.2-cp310-cp310-win_amd64.whl (40.1 MB)\n","        Using cached scipy-1.9.1-cp310-cp310-win_amd64.whl (38.6 MB)\n","        Using cached scipy-1.9.0-cp310-cp310-win_amd64.whl (38.6 MB)\n","        Using cached scipy-1.8.1-cp310-cp310-win_amd64.whl (36.9 MB)\n","      Building wheels for collected packages: numpy\n","        Building wheel for numpy (setup.py): started\n","        Building wheel for numpy (setup.py): finished with status 'error'\n","        error: subprocess-exited-with-error\n","      \n","        Ã— python setup.py bdist_wheel did not run successfully.\n","        â”‚ exit code: 1\n","        â•°â”€> [268 lines of output]\n","            Running from numpy source directory.\n","            blas_opt_info:\n","            blas_mkl_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries mkl_rt not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            blis_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries blis not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            openblas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries openblas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","            get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n","            customize GnuFCompiler\n","            Could not locate executable g77\n","            Could not locate executable f77\n","            customize IntelVisualFCompiler\n","            Could not locate executable ifort\n","            Could not locate executable ifl\n","            customize AbsoftFCompiler\n","            Could not locate executable f90\n","            customize CompaqVisualFCompiler\n","            Could not locate executable DF\n","            customize IntelItaniumVisualFCompiler\n","            Could not locate executable efl\n","            customize Gnu95FCompiler\n","            Could not locate executable gfortran\n","            Could not locate executable f95\n","            customize G95FCompiler\n","            Could not locate executable g95\n","            customize IntelEM64VisualFCompiler\n","            customize IntelEM64TFCompiler\n","            Could not locate executable efort\n","            Could not locate executable efc\n","            customize PGroupFlangCompiler\n","            Could not locate executable flang\n","            don't know how to compile Fortran code on platform 'nt'\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_blas_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_blas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_blas_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_blas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            accelerate_info:\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:690: UserWarning:\n","                Optimized (vendor) Blas libraries are not found.\n","                Falls back to netlib Blas library which has worse performance.\n","                A better performance should be easily gained by switching\n","                Blas library.\n","              self.calc_info()\n","            blas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries blas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:690: UserWarning:\n","                Blas (http://www.netlib.org/blas/) libraries not found.\n","                Directories to search for the libraries can be specified in the\n","                numpy/distutils/site.cfg file (section [blas]) or by setting\n","                the BLAS environment variable.\n","              self.calc_info()\n","            blas_src_info:\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:690: UserWarning:\n","                Blas (http://www.netlib.org/blas/) sources not found.\n","                Directories to search for the sources can be specified in the\n","                numpy/distutils/site.cfg file (section [blas_src]) or by setting\n","                the BLAS_SRC environment variable.\n","              self.calc_info()\n","              NOT AVAILABLE\n","      \n","            'svnversion' is not recognized as an internal or external command,\n","            operable program or batch file.\n","            non-existing path in 'numpy\\\\distutils': 'site.cfg'\n","            lapack_opt_info:\n","            lapack_mkl_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries mkl_rt not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            openblas_lapack_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries openblas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            openblas_clapack_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries openblas,lapack not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            flame_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries flame not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas,tatlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas,tatlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas,tatlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas,satlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas,satlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas,satlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_3_10_info'>\n","              NOT AVAILABLE\n","      \n","            atlas_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_threads_info'>\n","              NOT AVAILABLE\n","      \n","            atlas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_info'>\n","              NOT AVAILABLE\n","      \n","            lapack_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:1712: UserWarning:\n","                Lapack (http://www.netlib.org/lapack/) libraries not found.\n","                Directories to search for the libraries can be specified in the\n","                numpy/distutils/site.cfg file (section [lapack]) or by setting\n","                the LAPACK environment variable.\n","              if getattr(self, '_calc_info_{}'.format(lapack))():\n","            lapack_src_info:\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:1712: UserWarning:\n","                Lapack (http://www.netlib.org/lapack/) sources not found.\n","                Directories to search for the sources can be specified in the\n","                numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n","                the LAPACK_SRC environment variable.\n","              if getattr(self, '_calc_info_{}'.format(lapack))():\n","              NOT AVAILABLE\n","      \n","            C:\\Python310\\lib\\site-packages\\setuptools\\_distutils\\dist.py:257: UserWarning: Unknown distribution option: 'define_macros'\n","              warnings.warn(msg)\n","            running bdist_wheel\n","            running build\n","            running config_cc\n","            unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n","            running config_fc\n","            unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n","            running build_src\n","            build_src\n","            building py_modules sources\n","            creating build\n","            creating build\\src.win-amd64-3.1\n","            creating build\\src.win-amd64-3.1\\numpy\n","            creating build\\src.win-amd64-3.1\\numpy\\distutils\n","            building library \"npymath\" sources\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n","            [end of output]\n","      \n","        note: This error originates from a subprocess, and is likely not a problem with pip.\n","        ERROR: Failed building wheel for numpy\n","        Running setup.py clean for numpy\n","        error: subprocess-exited-with-error\n","      \n","        Ã— python setup.py clean did not run successfully.\n","        â”‚ exit code: 1\n","        â•°â”€> [10 lines of output]\n","            Running from numpy source directory.\n","      \n","            `setup.py clean` is not supported, use one of the following instead:\n","      \n","              - `git clean -xdf` (cleans all files)\n","              - `git clean -Xdf` (cleans all versioned files, doesn't touch\n","                                  files that aren't checked into the git repo)\n","      \n","            Add `--force` to your command to use it anyway if you must (unsupported).\n","      \n","            [end of output]\n","      \n","        note: This error originates from a subprocess, and is likely not a problem with pip.\n","        ERROR: Failed cleaning build dir for numpy\n","      Failed to build numpy\n","      WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","      Installing collected packages: wheel, setuptools, numpy, Cython, scipy\n","        Running setup.py install for numpy: started\n","        Running setup.py install for numpy: finished with status 'error'\n","        error: subprocess-exited-with-error\n","      \n","        Ã— Running setup.py install for numpy did not run successfully.\n","        â”‚ exit code: 1\n","        â•°â”€> [275 lines of output]\n","            Running from numpy source directory.\n","      \n","            Note: if you need reliable uninstall behavior, then install\n","            with pip instead of using `setup.py install`:\n","      \n","              - `pip install .`       (from a git repo or downloaded source\n","                                       release)\n","              - `pip install numpy`   (last NumPy release on PyPi)\n","      \n","      \n","            blas_opt_info:\n","            blas_mkl_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries mkl_rt not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            blis_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries blis not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            openblas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries openblas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","            get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n","            customize GnuFCompiler\n","            Could not locate executable g77\n","            Could not locate executable f77\n","            customize IntelVisualFCompiler\n","            Could not locate executable ifort\n","            Could not locate executable ifl\n","            customize AbsoftFCompiler\n","            Could not locate executable f90\n","            customize CompaqVisualFCompiler\n","            Could not locate executable DF\n","            customize IntelItaniumVisualFCompiler\n","            Could not locate executable efl\n","            customize Gnu95FCompiler\n","            Could not locate executable gfortran\n","            Could not locate executable f95\n","            customize G95FCompiler\n","            Could not locate executable g95\n","            customize IntelEM64VisualFCompiler\n","            customize IntelEM64TFCompiler\n","            Could not locate executable efort\n","            Could not locate executable efc\n","            customize PGroupFlangCompiler\n","            Could not locate executable flang\n","            don't know how to compile Fortran code on platform 'nt'\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_blas_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_blas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_blas_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_blas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            accelerate_info:\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:690: UserWarning:\n","                Optimized (vendor) Blas libraries are not found.\n","                Falls back to netlib Blas library which has worse performance.\n","                A better performance should be easily gained by switching\n","                Blas library.\n","              self.calc_info()\n","            blas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries blas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:690: UserWarning:\n","                Blas (http://www.netlib.org/blas/) libraries not found.\n","                Directories to search for the libraries can be specified in the\n","                numpy/distutils/site.cfg file (section [blas]) or by setting\n","                the BLAS environment variable.\n","              self.calc_info()\n","            blas_src_info:\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:690: UserWarning:\n","                Blas (http://www.netlib.org/blas/) sources not found.\n","                Directories to search for the sources can be specified in the\n","                numpy/distutils/site.cfg file (section [blas_src]) or by setting\n","                the BLAS_SRC environment variable.\n","              self.calc_info()\n","              NOT AVAILABLE\n","      \n","            'svnversion' is not recognized as an internal or external command,\n","            operable program or batch file.\n","            non-existing path in 'numpy\\\\distutils': 'site.cfg'\n","            lapack_opt_info:\n","            lapack_mkl_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries mkl_rt not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            openblas_lapack_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries openblas not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            openblas_clapack_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries openblas,lapack not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            flame_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries flame not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas,tatlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas,tatlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries tatlas,tatlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n","              NOT AVAILABLE\n","      \n","            atlas_3_10_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas,satlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas,satlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries satlas,satlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_3_10_info'>\n","              NOT AVAILABLE\n","      \n","            atlas_threads_info:\n","            Setting PTATLAS=ATLAS\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries ptf77blas,ptcblas,atlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_threads_info'>\n","              NOT AVAILABLE\n","      \n","            atlas_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in C:\\Python310\\lib\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in C:\\\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack_atlas not found in C:\\Python310\\libs\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries f77blas,cblas,atlas not found in C:\\Python310\\libs\n","            <class 'numpy.distutils.system_info.atlas_info'>\n","              NOT AVAILABLE\n","      \n","            lapack_info:\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            customize MSVCCompiler\n","              libraries lapack not found in ['C:\\\\Python310\\\\lib', 'C:\\\\', 'C:\\\\Python310\\\\libs']\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:1712: UserWarning:\n","                Lapack (http://www.netlib.org/lapack/) libraries not found.\n","                Directories to search for the libraries can be specified in the\n","                numpy/distutils/site.cfg file (section [lapack]) or by setting\n","                the LAPACK environment variable.\n","              if getattr(self, '_calc_info_{}'.format(lapack))():\n","            lapack_src_info:\n","              NOT AVAILABLE\n","      \n","            C:\\Users\\dantu\\AppData\\Local\\Temp\\pip-install-fsgjhjvm\\numpy_f61d1f87886e4209a5ddb4082a3ed2cf\\numpy\\distutils\\system_info.py:1712: UserWarning:\n","                Lapack (http://www.netlib.org/lapack/) sources not found.\n","                Directories to search for the sources can be specified in the\n","                numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n","                the LAPACK_SRC environment variable.\n","              if getattr(self, '_calc_info_{}'.format(lapack))():\n","              NOT AVAILABLE\n","      \n","            C:\\Python310\\lib\\site-packages\\setuptools\\_distutils\\dist.py:257: UserWarning: Unknown distribution option: 'define_macros'\n","              warnings.warn(msg)\n","            running install\n","            C:\\Python310\\lib\\site-packages\\setuptools\\command\\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n","              warnings.warn(\n","            running build\n","            running config_cc\n","            unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n","            running config_fc\n","            unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n","            running build_src\n","            build_src\n","            building py_modules sources\n","            building library \"npymath\" sources\n","            No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n","            error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n","            [end of output]\n","      \n","        note: This error originates from a subprocess, and is likely not a problem with pip.\n","      error: legacy-install-failure\n","      \n","      Ã— Encountered error while trying to install package.\n","      â•°â”€> numpy\n","      \n","      note: This is an issue with the package mentioned above, not pip.\n","      hint: See above for output from the failure.\n","      WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","      WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","      WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","      [end of output]\n","  \n","  note: This error originates from a subprocess, and is likely not a problem with pip.\n","error: subprocess-exited-with-error\n","\n","× pip subprocess to install build dependencies did not run successfully.\n","│ exit code: 1\n","╰─> See above for output.\n","\n","note: This error originates from a subprocess, and is likely not a problem with pip.\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n"]},{"name":"stdout","output_type":"stream","text":["Requirement already satisfied: xgboost==1.3.3 in c:\\python310\\lib\\site-packages (1.3.3)\n","Requirement already satisfied: numpy in c:\\python310\\lib\\site-packages (from xgboost==1.3.3) (1.23.4)\n","Requirement already satisfied: scipy in c:\\python310\\lib\\site-packages (from xgboost==1.3.3) (1.9.2)\n"]},{"name":"stderr","output_type":"stream","text":["WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n"]},{"name":"stdout","output_type":"stream","text":["Requirement already satisfied: lightgbm==3.1.1 in c:\\python310\\lib\\site-packages (3.1.1)\n","Requirement already satisfied: wheel in c:\\python310\\lib\\site-packages (from lightgbm==3.1.1) (0.37.1)\n","Requirement already satisfied: scipy in c:\\python310\\lib\\site-packages (from lightgbm==3.1.1) (1.9.2)\n","Requirement already satisfied: numpy in c:\\python310\\lib\\site-packages (from lightgbm==3.1.1) (1.23.4)\n","Requirement already satisfied: scikit-learn!=0.22.0 in c:\\python310\\lib\\site-packages (from lightgbm==3.1.1) (1.1.2)\n","Requirement already satisfied: joblib>=1.0.0 in c:\\python310\\lib\\site-packages (from scikit-learn!=0.22.0->lightgbm==3.1.1) (1.2.0)\n","Requirement already satisfied: threadpoolctl>=2.0.0 in c:\\python310\\lib\\site-packages (from scikit-learn!=0.22.0->lightgbm==3.1.1) (3.1.0)\n"]},{"name":"stderr","output_type":"stream","text":["WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n","WARNING: Ignoring invalid distribution -ip (c:\\python310\\lib\\site-packages)\n"]}],"source":["!pip install ibm-watson-machine-learning \n","!pip install -U autoai-libs==1.12.7 \n","!pip install -U 'lale>=0.5.1,<0.6' \n","!pip install -U scikit-learn==0.23.2 \n","!pip install -U xgboost==1.3.3 \n","!pip install -U lightgbm==3.1.1"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"configuration\"></a>\n","# Experiment configuration"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["<a id=\"metadata\"></a>\n","## Experiment metadata\n","This cell defines the metadata for the experiment, including: training_data_reference, training_result_reference, experiment_metadata."]},{"cell_type":"code","execution_count":4,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[{"ename":"ModuleNotFoundError","evalue":"No module named 'ibm_watson_machine_learning'","output_type":"error","traceback":["\u001b[1;31m---------------------------------------------------------------------------\u001b[0m","\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)","Cell \u001b[1;32mIn [4], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mibm_watson_machine_learning\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mhelpers\u001b[39;00m \u001b[39mimport\u001b[39;00m DataConnection\n\u001b[0;32m      2\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mibm_watson_machine_learning\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mhelpers\u001b[39;00m \u001b[39mimport\u001b[39;00m S3Connection, S3Location\n\u001b[0;32m      4\u001b[0m training_data_reference \u001b[39m=\u001b[39m [DataConnection(\n\u001b[0;32m      5\u001b[0m     connection\u001b[39m=\u001b[39mS3Connection(\n\u001b[0;32m      6\u001b[0m         api_key\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39m8MGQXLn_Vv9er3Ww4RuBIzZJd-7Ir8Xz4IKFgcc-E2X2\u001b[39m\u001b[39m'\u001b[39m,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     13\u001b[0m     )),\n\u001b[0;32m     14\u001b[0m ]\n","\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'ibm_watson_machine_learning'"]}],"source":["from ibm_watson_machine_learning.helpers import DataConnection\n","from ibm_watson_machine_learning.helpers import S3Connection, S3Location\n","\n","training_data_reference = [DataConnection(                                                                      \n","    connection=S3Connection(\n","        api_key='8MGQXLn_Vv9er3Ww4RuBIzZJd-7Ir8Xz4IKFgcc-E2X2',\n","        auth_endpoint='https://iam.bluemix.net/oidc/token/',\n","        endpoint_url='https://s3.eu-geo.objectstorage.softlayer.net'    \n","    ),\n","        location=S3Location(\n","        bucket='telcoconsumerchurn-donotdelete-pr-z6aanmrxdbpqcg',\n","        path='Telco-Customer-Churn.csv'\n","    )),\n","]\n","training_result_reference = DataConnection(\n","    connection=S3Connection(\n","        api_key='8MGQXLn_Vv9er3Ww4RuBIzZJd-7Ir8Xz4IKFgcc-E2X2',\n","        auth_endpoint='https://iam.bluemix.net/oidc/token/',\n","        endpoint_url='https://s3.eu-geo.objectstorage.softlayer.net'\n","    ),\n","    location=S3Location(\n","        bucket='telcoconsumerchurn-donotdelete-pr-z6aanmrxdbpqcg',\n","        path='auto_ml/8333b8d1-d61f-4781-a97c-289ffcefaa1c/wml_data/4ec64c50-6f28-4e7e-9448-f3e42734625a/data/automl',\n","        model_location='auto_ml/8333b8d1-d61f-4781-a97c-289ffcefaa1c/wml_data/4ec64c50-6f28-4e7e-9448-f3e42734625a/data/automl/pre_hpo_d_output/Pipeline1/model.pickle',\n","        training_status='auto_ml/8333b8d1-d61f-4781-a97c-289ffcefaa1c/wml_data/4ec64c50-6f28-4e7e-9448-f3e42734625a/training-status.json'\n","    ))"]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["experiment_metadata = dict(\n","   prediction_type='classification', \n","   prediction_column='Churn',\n","   holdout_size=0.1,\n","   scoring='accuracy',\n","   csv_separator=',',\n","   random_state=33,\n","   max_number_of_estimators=2,\n","   training_data_reference=training_data_reference,\n","   training_result_reference=training_result_reference,\n","   deployment_url='https://eu-gb.ml.cloud.ibm.com',\n","   project_id='992cbd02-a9bd-4ae0-a700-7eef966a4da2',\n","   positive_label='Yes',\n","   drop_duplicates=True\n",")"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"connection\"></a>\n","## Watson Machine Learning connection\n","\n","This cell defines the credentials required to work with the Watson Machine Learning service.\n","\n","**Action** Please provide IBM Cloud apikey following [docs](https://cloud.ibm.com/docs/account?topic=account-userapikey)."]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["api_key = 'PUT_YOUR_KEY_HERE'"]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["wml_credentials = {\n","    \"apikey\": api_key,\n","    \"url\": experiment_metadata['deployment_url']\n","}"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"work\"></a>\n","\n","\n","# Working with completed AutoAI experiment\n","\n","This cell imports the pipelines generated for the experiment so they can be compared to find the optimal pipeline to save as a model."]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"get\"></a>\n","\n","\n","## Get fitted AutoAI optimizer"]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[{"name":"stdout","output_type":"stream","text":["DEPRECATED!! Python 3.6 framework is deprecated and will be removed on Jan 20th, 2021. It will be read-only mode starting Nov 20th, 2020. i.e you won't be able to create new assets using this client. Use Python 3.7 instead. For details, see https://dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/pm_service_supported_frameworks.html\n"]},{"ename":"WMLClientError","evalue":"Error during getting IAM Token.\nReason: <Response [400]>","output_type":"error","traceback":["\u001b[1;31m---------------------------------------------------------------------------\u001b[0m","\u001b[1;31mWMLClientError\u001b[0m                            Traceback (most recent call last)","\u001b[1;32m<ipython-input-13-c77c63c9352e>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mibm_watson_machine_learning\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexperiment\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mAutoAI\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[0mpipeline_optimizer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mAutoAI\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mwml_credentials\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mproject_id\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mexperiment_metadata\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'project_id'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mruns\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_optimizer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmetadata\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mexperiment_metadata\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[1;32mC:\\Users\\dantu\\anaconda3\\envs\\yields\\lib\\site-packages\\ibm_watson_machine_learning\\experiment\\autoai\\autoai.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, wml_credentials, project_id, space_id)\u001b[0m\n\u001b[0;32m    102\u001b[0m                 self._workspace = WorkSpace(wml_credentials=wml_credentials.copy(),\n\u001b[0;32m    103\u001b[0m                                             \u001b[0mproject_id\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mproject_id\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 104\u001b[1;33m                                             space_id=space_id)\n\u001b[0m\u001b[0;32m    105\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    106\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mproject_id\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_workspace\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mproject_id\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n","\u001b[1;32mC:\\Users\\dantu\\anaconda3\\envs\\yields\\lib\\site-packages\\ibm_watson_machine_learning\\workspace\\workspace.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, wml_credentials, project_id, space_id)\u001b[0m\n\u001b[0;32m     72\u001b[0m         \u001b[1;31m# --- end note\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     73\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 74\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwml_client\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mAPIClient\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mwml_credentials\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwml_credentials\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     75\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     76\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mwml_credentials\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'instance_id'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n","\u001b[1;32mC:\\Users\\dantu\\anaconda3\\envs\\yields\\lib\\site-packages\\ibm_watson_machine_learning\\client.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, wml_credentials, project_id)\u001b[0m\n\u001b[0;32m    304\u001b[0m                     \"For details, see https://www.ibm.com/support/knowledgecenter/SSQNUZ_current/wsj/analyze-data/pm_service_supported_frameworks.html\")\n\u001b[0;32m    305\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 306\u001b[1;33m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mservice_instance\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mServiceInstanceNewPlan\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    307\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvolumes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mVolume\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    308\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n","\u001b[1;32mC:\\Users\\dantu\\anaconda3\\envs\\yields\\lib\\site-packages\\ibm_watson_machine_learning\\instance_new_plan.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, client)\u001b[0m\n\u001b[0;32m     47\u001b[0m                                                  \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_client\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mICP_PLATFORM_SPACES\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     48\u001b[0m                                                  )\n\u001b[1;32m---> 49\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_client\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwml_token\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_create_token\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     50\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_logger\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0minfo\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mu'Successfully prepared token: '\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_client\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwml_token\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     51\u001b[0m         \u001b[1;31m# ml_repository_client is initialized in repo\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n","\u001b[1;32mC:\\Users\\dantu\\anaconda3\\envs\\yields\\lib\\site-packages\\ibm_watson_machine_learning\\instance_new_plan.py\u001b[0m in \u001b[0;36m_create_token\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    222\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_client\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mICP_PLATFORM_SPACES\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    223\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_client\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_is_IAM\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 224\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_IAM_token\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    225\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    226\u001b[0m                 \u001b[1;32mraise\u001b[0m \u001b[0mWMLClientError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'apikey for IAM token is not provided in credentials for the client.'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n","\u001b[1;32mC:\\Users\\dantu\\anaconda3\\envs\\yields\\lib\\site-packages\\ibm_watson_machine_learning\\instance_new_plan.py\u001b[0m in \u001b[0;36m_get_IAM_token\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    282\u001b[0m             \u001b[0mtoken\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mu'access_token'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    283\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 284\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mWMLClientError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mu'Error during getting IAM Token.'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    285\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mtoken\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n","\u001b[1;31mWMLClientError\u001b[0m: Error during getting IAM Token.\nReason: <Response [400]>"]}],"source":["from ibm_watson_machine_learning.experiment import AutoAI\n","\n","pipeline_optimizer = AutoAI(wml_credentials, project_id=experiment_metadata['project_id']).runs.get_optimizer(metadata=experiment_metadata)"]},{"cell_type":"markdown","metadata":{},"source":["Use `get_params()`- to retrieve configuration parameters."]},{"cell_type":"code","execution_count":6,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[{"ename":"NameError","evalue":"name 'pipeline_optimizer' is not defined","output_type":"error","traceback":["\u001b[1;31m---------------------------------------------------------------------------\u001b[0m","\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)","Cell \u001b[1;32mIn [6], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m pipeline_optimizer\u001b[39m.\u001b[39mget_params()\n","\u001b[1;31mNameError\u001b[0m: name 'pipeline_optimizer' is not defined"]}],"source":["pipeline_optimizer.get_params()"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["<a id=\"comparison\"></a>\n","## Pipelines comparison\n","\n","Use the `summary()` method to list trained pipelines and evaluation metrics information in\n","the form of a Pandas DataFrame. You can use the DataFrame to compare all discovered pipelines and select the one you like for further testing."]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["summary = pipeline_optimizer.summary()\n","best_pipeline_name = list(summary.index)[0]\n","summary"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"get_pipeline\"></a>\n","### Get pipeline as scikit-learn pipeline model\n","\n","After you compare the pipelines, download and save a scikit-learn pipeline model object from the\n","AutoAI training job.\n","\n","**Tip:** If you want to get a specific pipeline you need to pass the pipeline name in:\n","```\n","pipeline_optimizer.get_pipeline(pipeline_name=pipeline_name)\n","```"]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["pipeline_model = pipeline_optimizer.get_pipeline()"]},{"cell_type":"markdown","metadata":{},"source":["Next, check features importance for selected pipeline."]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["pipeline_optimizer.get_pipeline_details()['features_importance']"]},{"cell_type":"markdown","metadata":{},"source":["**Tip:** If you want to check all model evaluation metrics-details, use:\n","```\n","pipeline_optimizer.get_pipeline_details()\n","```"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"inspect_pipeline\"></a>\n","## Inspect pipeline"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"visualize\"></a>\n","### Visualize pipeline model\n","\n","Preview pipeline model stages as a graph. Each node's name links to a detailed description of the stage.\n"]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["pipeline_model.visualize()"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"preview\"></a>\n","### Preview pipeline model as python code\n","In the next cell, you can preview the saved pipeline model as a python code.  \n","You will be able to review the exact steps used to create the model.\n","\n","**Note:** If you want to get sklearn representation add following parameter to `pretty_print` call: `astype='sklearn'`."]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["pipeline_model.pretty_print(combinators=False, ipython_display=True)"]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"scoring\"></a>\n","## Deploy and Score\n","\n","In this section you will learn how to deploy and score the model as a web service."]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"working_spaces\"></a>\n","### Working with spaces\n","\n","In this section you will specify a deployment space for organizing the assets for deploying and scoring the model. If you do not have an existing space, you can use [Deployment Spaces Dashboard](https://dataplatform.cloud.ibm.com/ml-runtime/spaces?context=cpdaas) to create a new space, following these steps:\n","\n","- Click **New Deployment Space**.\n","- Create an empty space.\n","- Select Cloud Object Storage.\n","- Select Watson Machine Learning instance and press **Create**.\n","- Copy `space_id` and paste it below.\n","\n","**Tip**: You can also use the SDK to prepare the space for your work. Learn more [here](https://github.com/IBM/watson-machine-learning-samples/blob/master/notebooks/python_sdk/instance-management/Space%20management.ipynb).\n","\n","**Action**: assign or update space ID below"]},{"cell_type":"markdown","metadata":{},"source":["### Deployment creation"]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["target_space_id = \"PUT_YOUR_TARGET_SPACE_ID_HERE\"\n","\n","from ibm_watson_machine_learning.deployment import WebService\n","service = WebService(source_wml_credentials=wml_credentials,\n","                     target_wml_credentials=wml_credentials,\n","                     source_project_id=experiment_metadata['project_id'],\n","                     target_space_id=target_space_id)\n","service.create(\n","model=best_pipeline_name,\n","metadata=experiment_metadata,\n","deployment_name='Best_pipeline_webservice'\n",")"]},{"cell_type":"markdown","metadata":{},"source":["Use the `print` method for the deployment object to show basic information about the service: "]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["print(service)"]},{"cell_type":"markdown","metadata":{},"source":["To show all available information about the deployment use the `.get_params()` method:"]},{"cell_type":"code","execution_count":null,"metadata":{"pycharm":{"name":"#%%\n"}},"outputs":[],"source":["service.get_params()"]},{"cell_type":"markdown","metadata":{},"source":["### Scoring of webservice\n","You can make scoring request by calling `score()` on the deployed pipeline."]},{"cell_type":"markdown","metadata":{},"source":["If you want to work with the web service in an external Python application,follow these steps to retrieve the service object:\n","\n"," - Initialize the service by `service = WebService(wml_credentials)`\n"," - Get deployment_id by `service.list()` method\n"," - Get webservice object by `service.get('deployment_id')` method\n","\n","After that you can call `service.score()` method."]},{"cell_type":"markdown","metadata":{},"source":["### Deleting deployment\n","<a id=\"cleanup\"></a>\n","You can delete the existing deployment by calling the `service.delete()` command.\n","To list the existing web services, use `service.list()`."]},{"cell_type":"markdown","metadata":{},"source":["<a id=\"run\"></a>\n","\n","## Running AutoAI experiment with Python SDK"]},{"cell_type":"markdown","metadata":{},"source":["If you want to run AutoAI experiment using python API follow up the steps decribed below. The experiment settings were generated basing on parameters set on UI.\n"," - Go to your COS dashboard.\n"," - In Service credentials tab, click New Credential.\n"," - Add the inline configuration parameter: `{“HMAC”:true}`, click Add.\n","This configuration parameter adds the following section to the instance credentials, (for use later in this notebook):\n","```\n","cos_hmac_keys”: {\n","      “access_key_id”: “***“,\n","      “secret_access_key”: “***”\n"," }\n"," ```\n","\n","**Action:** Please provide cos credentials in following cells.\n","\n","- Use provided markdown cells to run code.\n","\n"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%%\n"}},"source":["```\n","from ibm_watson_machine_learning.experiment import AutoAI\n","\n","experiment = AutoAI(wml_credentials, project_id=experiment_metadata['project_id'])\n","```"]},{"cell_type":"markdown","metadata":{},"source":["```\n","#@hidden_cell\n","cos_hmac_keys = {\n","    \"access_key_id\": \"PLACE_YOUR_ACCESS_KEY_ID_HERE\",\n","    \"secret_access_key\": \"PLACE_YOUR_SECRET_ACCESS_KEY_HERE\"\n","  }\n","  \n","cos_api_key = \"PLACE_YOUR_API_KEY_HERE\"\n","OPTIMIZER_NAME = 'custom_name'\n","```"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["The experiment settings were generated basing on parameters set on UI."]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% raw\n"}},"source":["```\n","from ibm_watson_machine_learning.helpers import DataConnection\n","from ibm_watson_machine_learning.helpers import S3Connection, S3Location\n","\n","training_data_reference = [DataConnection(\n","    connection=S3Connection(\n","        api_key=cos_api_key,\n","        auth_endpoint='https://iam.bluemix.net/oidc/token/',\n","        endpoint_url='https://s3.eu-geo.objectstorage.softlayer.net',\n","        access_key_id = cos_hmac_keys['access_key_id'],\n","        secret_access_key = cos_hmac_keys['secret_access_key']\n","    ),\n","        location=S3Location(\n","        bucket='telcoconsumerchurn-donotdelete-pr-z6aanmrxdbpqcg',\n","        path='Telco-Customer-Churn.csv'\n","    )),\n","]\n","from ibm_watson_machine_learning.helpers import S3Connection, S3Location\n","training_result_reference = DataConnection(\n","    connection=S3Connection(\n","        api_key=cos_api_key,\n","        auth_endpoint='https://iam.bluemix.net/oidc/token/',\n","        endpoint_url='https://s3.eu-geo.objectstorage.softlayer.net',\n","        access_key_id = cos_hmac_keys['access_key_id'],\n","        secret_access_key = cos_hmac_keys['secret_access_key']\n","    ),\n","    location=S3Location(\n","        bucket='telcoconsumerchurn-donotdelete-pr-z6aanmrxdbpqcg',\n","        path='auto_ml/8333b8d1-d61f-4781-a97c-289ffcefaa1c/wml_data/4ec64c50-6f28-4e7e-9448-f3e42734625a/data/automl',\n","        model_location='auto_ml/8333b8d1-d61f-4781-a97c-289ffcefaa1c/wml_data/4ec64c50-6f28-4e7e-9448-f3e42734625a/data/automl/pre_hpo_d_output/Pipeline1/model.pickle',\n","        training_status='auto_ml/8333b8d1-d61f-4781-a97c-289ffcefaa1c/wml_data/4ec64c50-6f28-4e7e-9448-f3e42734625a/training-status.json'\n","    ))\n","```"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%%raw\n"}},"source":["```\n","pipeline_optimizer = experiment.optimizer(\n","    name=OPTIMIZER_NAME,\n","    prediction_type=experiment_metadata['prediction_type'],\n","    prediction_column=experiment_metadata['prediction_column'],\n","    scoring=experiment_metadata['scoring'],\n","    holdout_size=experiment_metadata['holdout_size'],\n","    csv_separator=experiment_metadata['csv_separator'],\n","    positive_label=experiment_metadata['positive_label'],\n","    drop_duplicates=experiment_metadata['drop_duplicates'])\n","```"]},{"cell_type":"markdown","metadata":{},"source":["```\n","pipeline_optimizer.fit(training_data_reference=training_data_reference,\n","                       training_results_reference=training_result_reference,\n","                       background_mode=False)\n","```"]}],"metadata":{"kernelspec":{"display_name":"Python 3.10.6 64-bit","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.6"},"metadata":{"interpreter":{"hash":"e89e97564c7e0902a6f16f4cb8367c2fe34761a76f42fdc137b35cb4e4628d6a"}},"pycharm":{"stem_cell":{"cell_type":"raw","metadata":{"collapsed":false},"source":["\n"]}},"vscode":{"interpreter":{"hash":"369f2c481f4da34e4445cda3fffd2e751bd1c4d706f27375911949ba6bb62e1c"}}},"nbformat":4,"nbformat_minor":1}
